compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -411,6 +405,27 @@ impl DropTree {
                     };
                     cfg.terminate(block, drop_node.data.source_info, terminator);
                 }
+                DropKind::ForLint => {
```

Comment:
I ended up adjusting this to act like `DropKind::Storage`, but inserting a `BackwardIncompatibleDropHint` instead of a `StorageDead`. Previously we'd insert a drop terminator here.

Reply:
Makes sense.